pkg/osvscanner/filter.go

```diff
@@ -0,0 +1,144 @@
+package osvscanner
+
+import (
+	"fmt"
+
+	"github.com/google/osv-scanner/internal/config"
+	"github.com/google/osv-scanner/internal/imodels"
+	"github.com/google/osv-scanner/internal/imodels/results"
+	"github.com/google/osv-scanner/pkg/models"
+	"github.com/google/osv-scanner/pkg/reporter"
+)
+
+// filterUnscannablePackages removes packages that don't have enough information to be scanned
+// e,g, local packages that specified by path
+func filterUnscannablePackages(r reporter.Reporter, scanResults *results.ScanResults) {
+	packageResults := make([]imodels.PackageScanResult, 0, len(scanResults.PackageScanResults))
+	for _, psr := range scanResults.PackageScanResults {
+		p := psr.PackageInfo
+		switch {
```

Comment:
can we just use `if` statement here?

Reply:
Hmm it looks quite ugly/unreadable with if statements here, need to look something like:
```
		if (p.Ecosystem.Ecosystem == "" || p.Name == "" || p.Version == "") &&
			(p.Commit == "") {
			continue
		}
```

Reply:
hmm I see - I was curious if this is for some future extensibility...